viam-cartographer.go

```diff
@@ -637,7 +639,18 @@ func (cartoSvc *cartographerService) readDataOnInterval(ctx context.Context, lid
 
 func (cartoSvc *cartographerService) getNextDataPoint(ctx context.Context, lidar lidar.Lidar, c chan int) {
 	if _, err := dim2d.GetAndSaveData(ctx, cartoSvc.dataDirectory, lidar, cartoSvc.logger); err != nil {
-		cartoSvc.logger.Warn(err)
+		if strings.Contains(err.Error(), "reached end of dataset") {
+			if !jobsDone {
+				cartoSvc.logger.Warn(err)
+				_, err = os.Create(filepath.Join(cartoSvc.dataDirectory, "Jobs_Done.txt"))
+				if err != nil {
+					cartoSvc.logger.Warn(err)
+				}
+				jobsDone = true
```

Comment:
[note] This effectively makes [the loop that calls this function](https://github.com/viamrobotics/viam-cartographer/blob/8c0ff9f173bb3f49e57c9751d19b03dfa4277f46/viam-cartographer.go#L595-L604) spin continuously without doing any work, which will waste CPU.

However, it's not so different from what was happening before this patch (now we are simply not logging in each iteration of the loop). Also, we should only see "reached end of dataset" in offline mode, which will only run in the cloud and the job will exit very soon after we see that error.

So, I think it's okay to leave this as-is.

Reply:
yeah I did not want to dive deep to get cartographer to stop the data thread, especially with the move to full mod coming up